README.md

```diff
@@ -171,8 +177,82 @@ datalad containers-run \
         --output . \
         '{inputs}' '{outputs}' participant group -w workdir
 )
+
+```
+</details>
+
+## Walkthrough
+
+For users who are new to these components, we will walk through how
+these components are used together in a typical Yoda workflow.
+the steps
+
+```bash
+mkdir ~/my-experiments
+cd ~/my-experiments
+datalad create -d ds000003-qc -c text2git
+cd ds000003-qc
+```
+
+Datalad has created a new directory for our results, `ds000003-qc`.
+According to YODA principles, this dataset should also contain our input
+data, code, and anything else we need to run the analysis.
+
+Install the input dataset:
+
+```bash
+datalad install -d . -s https://github.com/ReproNim/ds000003-demo sourcedata
+```
+
+Next we install the `ReproNim/containers` collection.
+
+```bash
+datalad install -d . ///repronim/containers
 ```
 
+Now let's take a look at what we have.
+
+```bash
+/ds000003-qc # The root dataset contains everything
+    /sourcedata # we call it source, but it is actually 000003-demo
+    /repronim/containers # This is where our non-custom code lives
+```
+
+Now we can "freeze" a container or containers that we plan to use so
+anyone (including you) will know exactly which version was used. This is
+technically optional, but very helpful. (All that changes is a
+subproject commit hash.)
```

Comment:
I have remembered one "cons" for this IMHO worth noting. Please incorporate something like

```suggestion
subproject commit hash.)  A disadvantage though is that if you were to make
others to reproduce the results, you would need to share your clone of `///repronim/containers`
to share that updated state with frozen version in `.datalad/config`.
```

Reply:
Please re-read this section, I rewrote it with your ideas included.